framework/PostProcessors/FTImporter.py

```diff
@@ -0,0 +1,153 @@
+import copy
+import itertools
+from collections import OrderedDict
+#External Modules End-----------------------------------------------------------
+
+#Internal Modules---------------------------------------------------------------
+from .PostProcessor import PostProcessor
+from utils import InputData
+from utils import xmlUtils as xmlU
+from utils import utils
+from .FTstructure import FTstructure
+import Files
+import Runners
+#Internal Modules End-----------------------------------------------------------
+
+class FTImporter(PostProcessor):
+  """
+    This is the base class of the postprocessor that imports Fault-Trees (FTs) into RAVEN as a PointSet
+  """
+  def __init__(self, messageHandler):
+    """
+      Constructor
+      @ In, messageHandler, MessageHandler, message handler object
+      @ Out, None
+    """
+    PostProcessor.__init__(self, messageHandler)
+    self.printTag = 'POSTPROCESSOR FT IMPORTER'
+    self.FTFormat = None
+    self.allowedFormats = ['OpenPSA']
+
+    self.topEventID = None
+
+  @classmethod
+  def getInputSpecification(cls):
+    """
+      Method to get a reference to a class that specifies the input data for
+      class cls.
+      @ In, cls, the class for which we are retrieving the specification
+      @ Out, inputSpecification, InputData.ParameterInput, class to use for
+        specifying input of cls.
+    """
+    inputSpecification = super(FTImporter, cls).getInputSpecification()
+    inputSpecification.addSub(InputData.parameterInputFactory("fileFormat", contentType=InputData.StringType))
+    inputSpecification.addSub(InputData.parameterInputFactory("topEventID", contentType=InputData.StringType))
+    return inputSpecification
+
+  def initialize(self, runInfo, inputs, initDict) :
+    """
+      Method to initialize the pp.
+      @ In, runInfo, dict, dictionary of run info (e.g. working dir, etc)
+      @ In, inputs, list, list of inputs
+      @ In, initDict, dict, dictionary with initialization options
+      @ Out, None
+    """
+    PostProcessor.initialize(self, runInfo, inputs, initDict)
+
+  def _localReadMoreXML(self, xmlNode):
+    """
+      Function to read the portion of the xml input that belongs to this specialized class
+      and initialize some stuff based on the inputs got
+      @ In, xmlNode, xml.etree.Element, Xml element node
+      @ Out, None
+    """
+    paramInput = FTImporter.getInputSpecification()()
+    paramInput.parseNode(xmlNode)
+    self._handleInput(paramInput)
+
+  def _handleInput(self, paramInput):
+    """
+      Method that handles PostProcessor parameter input block.
+      @ In, paramInput, ParameterInput, the already parsed input.
+      @ Out, None
+    """
+    fileFormat = paramInput.findFirst('fileFormat')
+    self.fileFormat = fileFormat.value
+    if self.fileFormat not in self.allowedFormats:
+      self.raiseAnError(IOError, 'FTImporterPostProcessor Post-Processor ' + self.name + ', format ' + str(self.fileFormat) + ' : is not supported')
```

Comment:
if you use EunumType this check is not needed

Reply:
EnumType is used, and the above lines are removed.